apps/vc-api/src/vc-api/exchanges/exchange.service.ts

```diff
@@ -122,8 +124,17 @@ export class ExchangeService {
       } as PresentationSubmissionEntity
     } as TransactionEntity);
 
-    // TODO: react to validation errors
-    const validationErrors = await validate(body, { whitelist: true, forbidUnknownValues: true });
+    const validationErrors = await validate(body, {
+      whitelist: true,
+      forbidUnknownValues: true,
+      forbidNonWhitelisted: false // here we want properties not defined in the CallbackDto to be just stripped out and not sent to a callback endpoint
+    });
+
+    if (validationErrors.length > 0) {
+      this.logger.error('\n' + validationErrors.map((e) => e.toString()).join('\n\n'));
+      throw new Error(validationErrors.toString());
+    }
+
     callback?.forEach((callback) => {
       this.httpService.post(callback.url, body).subscribe({
         next: (v) => Logger.log(inspect(v)), // inspect used to replace circular references https://stackoverflow.com/a/18354289
```

Comment:
I know it's not clearly part of this change, but I think we should use the new logger instantiation here as well, right?
```suggestion
        next: (v) => this.logger.log(inspect(v)), // inspect used to replace circular references https://stackoverflow.com/a/18354289
```

Reply:
100% agree! thanks!